python/pyspark/sql/readwriter.py

```diff
@@ -395,6 +425,38 @@ def partitionBy(self, *cols):
         self._jwrite = self._jwrite.partitionBy(_to_seq(self._sqlContext._sc, cols))
         return self
 
+    @since(2.0)
+    def queryName(self, queryName):
+        """Specifies the name of the :class:`ContinuousQuery` that can be started with
+        :func:`startStream`. This name must be unique among all the currently active queries
+        in the associated SQLContext.
+
+        :param queryName: unique name for the query
+
+        >>> writer = sdf.write.queryName('streaming_query')
+        """
+        if not queryName or type(queryName) != str or len(queryName.strip()) == 0:
+            raise ValueError('The queryName must be a non-empty string. Got: %s' % queryName)
+        self._jwrite = self._jwrite.queryName(queryName)
+        return self
+
+    @since(2.0)
+    def trigger(self, trigger):
+        """Set the trigger for the stream query. If this is not set it will run the query as fast
+        as possible, which is equivalent to setting the trigger to ``ProcessingTime('0 seconds')``.
+
+        :param trigger: a :class:`Trigger`.
+
+        >>> from pyspark.sql.streaming import ProcessingTime
+        >>> # trigger the query for execution every 5 seconds
+        >>> writer = sdf.write.trigger(ProcessingTime('5 seconds'))
```

Comment:
I'm only asking named parameters or classes.  Named parameters sounds like less overhead to me since you don't have type safety anyway, but I'm torn between that and matching scala exactly.